### DISCUSSION TITLE
Finetuning facebook/wav2vec2-xls-r-2b

### DISCUSSION THREAD

**Discussion by Tirthankar-iiitb** [k2-fsa/icefall#1458] [General]
The requirement is to fine-tune the XLSR-2B pre-trained model for a new language/accent (Bhojpuri) using adapters. I have ~6 hrs of audio & transcriptions for the new language/accent. I want to use K2 for the same. Which recipe I can simulate? Any indicator to this will be very helpful. 

I can do similar fine-tuning using Huggingface (HF). But I need to use Sherpa for inferences which I cannot do on the HF model.

**Comment by marcoyang1998** [comment]
You may look at https://github.com/marcoyang1998/icefall/tree/finetune_hubert/egs/librispeech/ASR/finetune_hubert_transducer, it's a recipe for fine-tuning a HuBERT model.

Also, if you want to deploy a wav2vec2 model with Sherpa, you may find this useful (https://github.com/k2-fsa/sherpa/pull/198). Doing the fine-tuning in icefall is not necessary for deployment with Sherpa, as long as you have the model in the right format (torchscript, onnx etc.)

**Reply by Tirthankar-iiitb** [reply]
Thanks.

**Reply by Tirthankar-iiitb** [reply]
@marcoyang1998 - I was trying to run finetune_hubert_transducer (lhose version - '1.20.0.dev+git.0089643.clean') but faced with an error 'ImportError: cannot import name 'SingleCutSampler'. 

from asr_datamodule import LibriSpeechAsrDataModule
File "/root/icefall/egs/librispeech/tirtho_ASR/finetune_hubert_transducer/asr_datamodule.py", line 37, in <module>
from lhotse.dataset.sampling import SingleCutSampler
ImportError: cannot import name 'SingleCutSampler' from 'lhotse.dataset.sampling' (/root/.virtualenvs/newstream/lib/python3.8/site-packages/lhotse/dataset/sampling/__init__.py)

Any other library to be used with the latest Lhotse version?

Thanks.

**Reply by mohsen-goodarzi** [reply]
It seems that your icefall version is a bit old for your Lhotse.
More info:
https://github.com/lhotse-speech/lhotse/pull/546

**Reply by Tirthankar-iiitb** [reply]
Oh...ok. Thanks. Let me consolidate the versions.

**Reply by marcoyang1998** [reply]
Note that the recipe is quite old and is just for a reference @Tirthankar-iiitb

**Reply by Tirthankar-iiitb** [reply]
Ok thanks. Let me try finetune.sh for the same task with necessary changes. Incase you have another recipe (latest), please do share. Thanks again.
